packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -0,0 +1,295 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'theme.dart';
+
+const double _kDefaultTooltipBorderRadius = 2.0;
+const double _kDefaultTooltipHeight = 32.0;
+const EdgeDims _kDefaultTooltipPadding = const EdgeDims.symmetric(horizontal: 16.0);
+const double _kDefaultVerticalTooltipOffset = 24.0;
+const EdgeDims _kDefaultTooltipScreenEdgeMargin = const EdgeDims.all(10.0);
+const Duration _kDefaultTooltipFadeDuration = const Duration(milliseconds: 200);
+const Duration _kDefaultTooltipShowDuration = const Duration(seconds: 2);
+
+class Tooltip extends StatefulComponent {
+  Tooltip({
+    Key key,
+    this.message,
+    this.backgroundColor,
+    this.textColor,
+    this.style,
+    this.opacity: 0.9,
+    this.borderRadius: _kDefaultTooltipBorderRadius,
+    this.height: _kDefaultTooltipHeight,
+    this.padding: _kDefaultTooltipPadding,
+    this.verticalOffset: _kDefaultVerticalTooltipOffset,
+    this.screenEdgeMargin: _kDefaultTooltipScreenEdgeMargin,
+    this.preferBelow: true,
+    this.fadeDuration: _kDefaultTooltipFadeDuration,
+    this.showDuration: _kDefaultTooltipShowDuration,
+    this.child
+  }) : super(key: key) {
+    assert(message != null);
+    assert(opacity != null);
+    assert(borderRadius != null);
+    assert(height != null);
+    assert(padding != null);
+    assert(verticalOffset != null);
+    assert(screenEdgeMargin != null);
+    assert(preferBelow != null);
+    assert(fadeDuration != null);
+    assert(showDuration != null);
+  }
+
+  final String message;
+  final Color backgroundColor;
+  final Color textColor;
+  final TextStyle style;
+  final double opacity;
+  final double borderRadius;
+  final double height;
+  final EdgeDims padding;
+  final double verticalOffset;
+  final EdgeDims screenEdgeMargin;
+  final bool preferBelow;
+  final Duration fadeDuration;
+  final Duration showDuration;
+  final Widget child;
+
+  TooltipState createState() => new TooltipState();
+}
+
+class TooltipState extends State<Tooltip> {
+
+  Performance _performance;
+  OverlayEntry _entry;
+  Timer _timer;
+
+  void initState() {
+    super.initState();
+    _performance ??= new Performance(duration: config.fadeDuration)
```

Comment:
It's not needed. I moved this code from the showTooltip() method and forgot to update it.